source/standalone/tutorials/04_sensors/benchmark_cameras.py

```diff
@@ -0,0 +1,621 @@
+    camera_cls: type[Camera | TiledCamera],
+    camera_cfg: type[CameraCfg | TiledCameraCfg],
+    num_cams: int,
+    data_types: list[str],
+    height: int,
+    width: int,
+) -> Camera | TiledCamera | None:
+    """Generalized function to create a camera or tiled camera sensor."""
+    # Determine prim prefix based on the camera class
+    name = camera_cls.__name__
+
+    # Create the necessary prims
+    for idx in range(num_cams):
+        prim_utils.create_prim(f"/World/{name}_{idx:02d}", "Xform")
+
+    # If valid camera settings are provided, create the camera
+    if num_cams > 0 and len(data_types) > 0 and height > 0 and width > 0:
+        cfg = camera_cfg(
+            prim_path=f"/World/{name}_.*/{name}",
+            update_period=0,
+            height=height,
+            width=width,
+            data_types=data_types,
+            spawn=sim_utils.PinholeCameraCfg(
+                focal_length=24, focus_distance=400.0, horizontal_aperture=20.955, clipping_range=(0.1, 1e4)
+            ),
+        )
+        return camera_cls(cfg=cfg)
+    else:
+        return None
+
+
+def create_tiled_cameras(
+    num_cams: int = 2, data_types: list[str] = ["rgb", "depth"], height: int = 100, width: int = 120
+) -> TiledCamera | None:
+    """Defines the tiled camera sensor to add to the scene."""
+    return create_camera_base(TiledCamera, TiledCameraCfg, num_cams, data_types, height, width)
+
+
+def create_cameras(
+    num_cams: int = 2, data_types: list[str] = ["rgb", "distance_to_image_plane"], height: int = 100, width: int = 120
+) -> Camera | None:
+    """Defines the camera sensor to add to the scene."""
+    return create_camera_base(Camera, CameraCfg, num_cams, data_types, height, width)
+
+
+def create_ray_caster_cameras(
+    num_cams: int = 2,
+    data_types: list[str] = ["distance_to_image_plane"],
+    mesh_prim_paths: list[str] = ["/World/ground"],
+    height: int = 100,
+    width: int = 120,
+) -> RayCasterCamera | None:
+    for idx in range(num_cams):
+        prim_utils.create_prim(f"/World/RayCasterCamera_{idx:02d}/RayCaster", "Xform")
+
+    if num_cams > 0 and len(data_types) > 0 and height > 0 and width > 0:
+        cam_cfg = RayCasterCameraCfg(
+            prim_path="/World/RayCasterCamera_.*/RayCaster",
+            mesh_prim_paths=mesh_prim_paths,
+            update_period=0,
+            offset=RayCasterCameraCfg.OffsetCfg(pos=(0.0, 0.0, 0.0), rot=(1.0, 0.0, 0.0, 0.0)),
+            data_types=data_types,
+            debug_vis=False,
+            pattern_cfg=patterns.PinholeCameraPatternCfg(
+                focal_length=24.0,
+                horizontal_aperture=20.955,
+                height=480,
+                width=640,
+            ),
+        )
+        return RayCasterCamera(cfg=cam_cfg)
+
+    else:
+        return None
+
+
+def design_scene(
+    num_tiled_cams: int = 2,
+    num_standard_cams: int = 0,
+    num_ray_caster_cams: int = 0,
+    standard_camera_replicators: list[str] = ["rgb"],
```

Comment:
Not super critical here as we only call this method once, but I generally avoid default `list` arguments to functions.

Some more details here in case you're not aware: https://www.geeksforgeeks.org/default-arguments-in-python/